CT-1552 Update Business Units that can handle SAR cases

stephenrichards opened the pull request.

misaka commented on app/assets/javascripts/application.js:
```diff
@@ -12,4 +12,19 @@
+//= require_tree ./modules
+
+(function () {
+  'use strict';
```
Holy bejezus, ES just keeps getting hackier and hackier ... I hadn't even seen this one! Not since [Perl](http://perldoc.perl.org/strict.html), that is!

misaka commented on app/assets/javascripts/modules/ShowHideContent.js:
```diff
@@ -0,0 +1,26 @@
+moj.Modules.ShowHideContent = {
+  el: 'label:has(":radio, :checkbox")',
```
What does "el" mean here?

aliuk2012 replied:
element as a string value you will see as part of the cacheEls we convert that into a jQuery object so we can apply jQuery methods on it.

misaka replied:
Ya, noticed that afterwards. It'd be nice to have something a bit less obscure here, just generally speaking.

aliuk2012 replied:
Sure normally I would try but the thing is if I use "element" it would be confusing when you started looking at the bindEvents method where that uses jQuery's .each() method which uses 'element'

misaka replied:
Would `elementSelector` be too annoyingly long?

aliuk2012 commented on app/views/assignments/edit.html.slim:
```diff
@@ -4,11 +4,20 @@
   .column-two-thirds
     p.lede
-      span.visually-hidden = t('.subject')
+      span.visually-hidden#subject = t('.subject')
```
Why do we need subject as an ID?

misaka replied:
I don't see it  added to the SitePrisms so I'm not sure we do.

aliuk2012 commented on app/views/assignments/edit.html.slim:
```diff
@@ -4,11 +4,20 @@
   .column-two-thirds
     h2.form-label-bold = t('.request')
-    .wrapper
+    .wrapper#subject
```
invalid HTML! All IDs on a page should be unique.  If you need to refer to a couple elements on the page then a css class should be used.

misaka commented on app/controllers/assignments_controller.rb:
```diff
@@ -22,8 +22,10 @@ def create
-      redirect_to 'case#show', params: { id: params[:case_id] }
+    new_state = (assignment_params[:state] + '!').to_sym
+
+    if @assignment.send(new_state)
```
After futsing around with this for a bit, I realised this is all going to change anyway when we implement the new state stuff anyway. We can just ignore this for now.

stephenrichards committed:
CT-1511 Restrict linking of SAR cases

SAR cases may only be linked by the manager of that case, and only to other
SAR cases

stephenrichards committed:
CT-1552 Update Business Units that can process SARs

stephenrichards committed:
CT-1552 Merge develop

stephenrichards committed:
CT-1522 regenerate structure.sql

stephenrichards committed:
CT-1522 extend capybara timeout
